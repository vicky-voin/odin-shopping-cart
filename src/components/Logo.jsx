import { Handbag } from "lucide-react";
import styles from "./Logo.module.css";

export default function Logo() {
  return (
    <div className={styles.logoRoot}>
      <div className={styles.logoText}>every</div>
      <div className={styles.logoText}>thing</div>
      <div className={styles.logoText}>shop.</div>
      <div className={styles.logoImage}>
        <Handbag size={52}></Handbag>
      </div>
    </div>
  );
}
