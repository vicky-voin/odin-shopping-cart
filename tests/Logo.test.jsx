import { describe, it, expect } from "vitest";
import { screen, render } from "@testing-library/react";
import Logo from "../src/components/Logo";

describe("Logo component tests", () => {
  it("renders a medium sized logo when no size is specified", () => {
    const { container } = render(<Logo></Logo>);
    expect(container).toMatchSnapshot();
  });

  it("renders a medium logo when a large size is requested", () => {
    const { container } = render(<Logo size={"medium"}></Logo>);

    //I haven't found a reliable way to test non-inline styling,
    // so the test doesn't cover font size change.
    const img = container.querySelector("svg");
    expect(img).toBeInTheDocument();
    expect(img).toHaveAttribute("width", "42");
    expect(img).toHaveAttribute("height", "42");
  });

  it("renders a large logo when a medium size is requested", () => {
    const { container } = render(<Logo size={"large"}></Logo>);

    //I haven't found a reliable way to test non-inline styling,
    // so the test doesn't cover font size change.
    const img = container.querySelector("svg");
    expect(img).toBeInTheDocument();
    expect(img).toHaveAttribute("width", "128");
    expect(img).toHaveAttribute("height", "128");
  });
});
