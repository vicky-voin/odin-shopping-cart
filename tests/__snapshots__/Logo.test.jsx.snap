// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Logo component tests > renders a medium sized logo when no size is specified 1`] = `
<div>
  <div
    aria-label="Store logo"
    class="_logoRoot_6dcafc"
  >
    <div
      class="_logoText_6dcafc _medium_6dcafc"
    >
      every
    </div>
    <div
      class="_logoText_6dcafc _medium_6dcafc"
    >
      thing
    </div>
    <div
      class="_logoText_6dcafc _medium_6dcafc"
    >
      shop.
    </div>
    <div
      aria-hidden="true"
      class="_logoImage_6dcafc"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-handbag"
        fill="none"
        height="42"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="42"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M2.048 18.566A2 2 0 0 0 4 21h16a2 2 0 0 0 1.952-2.434l-2-9A2 2 0 0 0 18 8H6a2 2 0 0 0-1.952 1.566z"
        />
        <path
          d="M8 11V6a4 4 0 0 1 8 0v5"
        />
      </svg>
    </div>
  </div>
</div>
`;
